lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -978,6 +978,12 @@ public void onAuthUpdated(final String token, final boolean waitForResponse) thr
      */
     public void onAuthError(ErrorInfo errorInfo) {
         Log.i(TAG, String.format("onAuthError: (%d) %s", errorInfo.code, errorInfo.message));
+
+        if(errorInfo.code == 403) {
+            this.connection.state = ConnectionState.failed;
```

Comment:
I feel you should check if there is a better way to set the state. Generally we don't directly set the state. You can check for method `setState` on connection. 
- SetState is responsible for broadcasting state change to components that need to act upon state change

Reply:
Correct, just setting the state here will break stuff